tests/functional/test_glauth.py

```diff
@@ -22,5 +20,5 @@ def setUp(self) -> None:
 
     def test_install(self):
         """Validate snap install."""
-        self.assertTrue(Path("/snap/bin/glauth").exists())
-        self.assertEqual(subprocess.check_call(["/snap/bin/glauth", "--version"]), 0)
+        self.assertTrue(glauth.installed)
+        self.assertTrue(type(glauth.version), str)
```

Comment:
Hmm... I am not sure about this unit test. To cover or not to cover...

With how this test is currently, `glauth.version` could return anything as long as its type is string. For example, the following test would pass:

```python
self.assertTrue(type("yowzah"), str)
```

I don't think this is exactly what we want. I would prefer if we evaluated `glauth.version` against a specific version number and then just updated the test as new releases of the GLAuth snap are published to the snap store. What are your thoughts on this?

Reply:
Previously I had something like `is_installed()` which was a property for a class that I had at the time. When stripping away the class and sticking with the module, I wanted to keep something property-like and I used __getattr__ instead on the module which satisfied that. 

As for why I wrote the functional test this way, I saw this and I thought it was a good model to follow: https://github.com/jnsgruk/parca-operator/blob/main/tests/functional/test_parca.py

Reply:
Ah yes, okay. So the subtle bit of difference is that the referenced code is using this inside of a class vs at the module level. Seeing this as a reference, I think I understand why you went down this route. The only thing is that I think that this will get confusing to future you in that it is obscured at the module level. While I know there's an approved PEP for __getattr__ on the module, I don't think this is the type of scenario it should be used in. I think the __getattr__ at the module level should be removed since its not being used for lazy imports, handling deprecated bits, etc. It's used for syntactic sugar more than anything, which can be done via a class and will be a bit more clear to future you.

For the unittest itself, I'd say revert to just checking the command runs successfully. You'll never be able to validate the version string but you can test that the version command runs as expected.

Reply:
That makes sense, removed **getattr** and reverted the test as suggested.